src/main/scala/mesosphere/marathon/api/RestResource.scala

```diff
@@ -2,25 +2,35 @@ package mesosphere.marathon
 package api
 
 import java.net.URI
+import javax.ws.rs.container.AsyncResponse
 import javax.ws.rs.core.Response
 import javax.ws.rs.core.Response.{ ResponseBuilder, Status }
 
 import akka.http.scaladsl.model.StatusCodes
-import com.wix.accord._
+import com.wix.accord.{ Failure => ValidationFailure, Validator, Success => ValidationSuccess }
 import mesosphere.marathon.api.v2.Validation._
 import mesosphere.marathon.api.v2.json.Formats._
 import mesosphere.marathon.core.deployment.DeploymentPlan
 import mesosphere.marathon.state.{ PathId, Timestamp }
 import play.api.libs.json.JsonValidationError
 import play.api.libs.json.Json.JsValueWrapper
 import play.api.libs.json._
-
-import scala.concurrent.{ Await, Awaitable }
+import scala.concurrent.{ Await, Awaitable, ExecutionContext, Future }
+import scala.util.{ Failure, Success }
 
 trait RestResource extends JaxResource {
-
+  implicit val executionContext: ExecutionContext
   protected val config: MarathonConf
-
+  case class FailureResponse(response: Response) extends Throwable
+
+  def sendResponse(asyncResponse: AsyncResponse)(future: Future[Response]) = {
```

Comment:
How are we handling timeouts?

Reply:
Great question :) Will make sure there's a TODO to handle this before removing WIP status

Reply:
Default timeout behavior is correct: no server timeout. If the client is not configured to time out it will go on indefinitely.

Reply:
should we maybe put that ^^ to some comment there? someone else in the future might be asking the same question